core/src/main/java/com/linecorp/armeria/client/circuitbreaker/AbstractCircuitBreakerClientBuilder.java

```diff
@@ -22,24 +22,26 @@
 import com.google.common.base.MoreObjects;
 
 import com.linecorp.armeria.client.Client;
+import com.linecorp.armeria.common.Request;
 import com.linecorp.armeria.common.Response;
 import com.linecorp.armeria.common.annotation.Nullable;
+import com.linecorp.armeria.common.annotation.UnstableApi;
 
 /**
  * A skeletal builder implementation that builds a new {@link AbstractCircuitBreakerClient} or
  * its decorator function.
  *
+ * @param <I> the type of incoming {@link Request} of the {@link Client}
  * @param <O> the type of incoming {@link Response} of the {@link Client}
  */
-public abstract class AbstractCircuitBreakerClientBuilder<O extends Response> {
+public abstract class AbstractCircuitBreakerClientBuilder<I extends Request, O extends Response> {
 
     @Nullable
     private final CircuitBreakerRule rule;
-
     @Nullable
     private final CircuitBreakerRuleWithContent<O> ruleWithContent;
-
-    private CircuitBreakerMapping mapping = CircuitBreakerMapping.ofDefault();
+    private CircuitBreakerClientHandler<I> handler =
+            DefaultCircuitBreakerClientHandler.of(CircuitBreakerMapping.ofDefault());
```

Comment:
Should we pre-populate and reuse the `DefaultCircuitBreakerClientHandler` for the default mapping and the default `CircuitBreaker`?

Reply:
It's a generic handler so we can't precompute an instance (if the following is what you meant)

```
private static final DefaultCircuitBreakerClientHandler<I> DEFAULT_HANDLER =
        DefaultCircuitBreakerClientHandler.of(CircuitBreakerMapping.ofDefault());
private CircuitBreakerClientHandler<I> handler = DEFAULT_HANDLER;
```